go/store/nbs/journal.go

```diff
@@ -640,15 +640,21 @@ func (jm *journalManifest) UpdateGCGen(ctx context.Context, lastLock hash.Hash,
 
 	t1 := time.Now()
 	defer func() { stats.WriteManifestLatency.SampleTimeSince(t1) }()
-	checker := func(upstream, contents manifestContents) error {
-		if contents.gcGen == upstream.gcGen {
-			return errors.New("UpdateGCGen() must update the garbage collection generation")
-		} else if contents.root != upstream.root {
-			return errors.New("UpdateGCGen() cannot update the root")
-		}
+	return updateWithChecker(ctx, jm.dir, syncFlush, updateGCGenManifestCheck, lastLock, newContents, writeHook)
+}
+
+func updateGCGenManifestCheck(upstream, contents manifestContents) error {
+	if contents.root != upstream.root {
+		return errors.New("UpdateGCGen() cannot update the root")
```

Comment:
Is this always an indication of a bug, or is it sometimes expected? If the former, language encouraging the user to report it would be a good idea.